cmd/private-org-sync/main.go

```diff
@@ -461,34 +461,72 @@ func main() {
 		failOnNonexistentDst: o.failOnNonexistentDst,
 	}
 
-	var syncErrors []error
+	var errors []error
+	var locations []location
+
+	if len(o.WhitelistOptions.WhitelistConfig.Whitelist) > 0 {
+		whitelistedLocations, err := getWhitelistedLocations(o.WhitelistOptions.WhitelistConfig, syncer.git)
+		if err != nil {
+			errors = append(errors, err)
+		}
+		locations = append(locations, whitelistedLocations...)
+	}
+
 	callback := func(_ *api.ReleaseBuildConfiguration, repoInfo *config.Info) error {
-		syncer.logger = config.LoggerForInfo(*repoInfo)
-		source := location{
+		locations = append(locations, location{
 			org:    repoInfo.Org,
 			repo:   repoInfo.Repo,
 			branch: repoInfo.Branch,
-		}
+		})
+		return nil
+	}
+
+	if err := config.OperateOnCIOperatorConfigDir(o.configDir, o.makeFilter(callback)); err != nil {
+		errors = append(errors, err)
+	}
+
+	for _, source := range locations {
+		syncer.logger = config.LoggerForInfo(config.Info{
+			Org:    source.org,
+			Repo:   source.repo,
+			Branch: source.branch,
+		})
+
 		destination := source
 		destination.org = o.targetOrg
 		gitDir, err := syncer.makeGitDir(source.org, source.repo)
 		if err != nil {
-			syncErrors = append(syncErrors, fmt.Errorf("%s->%s: %v", source.String(), destination.String(), err))
-			return nil
+			errors = append(errors, fmt.Errorf("%s->%s: %v", source.String(), destination.String(), err))
+			continue
 		}
 
 		if err := syncer.mirror(gitDir, source, destination); err != nil {
-			syncErrors = append(syncErrors, fmt.Errorf("%s->%s: %v", source.String(), destination.String(), err))
+			errors = append(errors, fmt.Errorf("%s->%s: %v", source.String(), destination.String(), err))
 		}
-		return nil
 	}
 
-	callback = o.makeFilter(callback)
-	if err := config.OperateOnCIOperatorConfigDir(o.configDir, callback); err != nil || len(syncErrors) > 0 {
-		if err != nil {
-			syncErrors = append(syncErrors, err)
+	if len(errors) > 0 {
+		logrus.WithError(utilerrors.NewAggregate(errors)).Fatal("There were failures")
+	}
+}
+
+func getWhitelistedLocations(whitelistConfig config.WhitelistConfig, git gitFunc) ([]location, error) {
```

Comment:
If you just need the whitelist itself, can we pass that instead of the larger config object?